### PR TITLE
Put SSH keypair rotation on maintenance behind a feature gate

### PR DESCRIPTION
**Pull request by ialidzhikov**
/area quality
/kind bug

As there are still items to work on (https://github.com/gardener/gardener/issues/4395, https://github.com/gardener/gardener/issues/4396) and to give a better control to the gardener operators on the rollout of the feature, with @petersutter we decided to put the SSH keypair rotation on maintenance feature behind a feature gate in the gardener-controller-manager.

**Release note**:

```other operator
The SSH keypair rotation on maintenance window is now set behind a new alpha feature gate in gardener-controller-manager - `RotateSSHKeypairOnMaintenance `.
```

### CODE REVIEW

**Review by vpnachev — DISMISSED** [review]
/lgtm

File: pkg/controllermanager/controller/shoot/shoot_maintenance_control.go [hunk 1]
```diff
@@ -189,6 +191,10 @@ func (r *shootMaintenanceReconciler) reconcile(ctx context.Context, shoot *garde
 		shoot.Spec.Kubernetes.Version = *updatedKubernetesVersion
 	}
 
+	if hasMaintainNowAnnotation(shoot) {
+		delete(shoot.Annotations, v1beta1constants.GardenerOperation)
+	}
```
Comment (petersutter) [inline pkg/controllermanager/controller/shoot/shoot_maintenance_control.go:196]: why was this added / why is this needed?

Comment (ialidzhikov) [reply pkg/controllermanager/controller/shoot/shoot_maintenance_control.go:196]: I think this is needed for the case when RotateSSHKeypairOnMaintenance is disabled and the Shoot lastOperation is not in Succeeded state. Currently, gardener.cloud/operation=reconcile annotation is added only when the Shoot lastOperation is Succeeded. It is actually something that was removed by https://github.com/gardener/gardener/pull/4224.

**Comment by ialidzhikov** [comment]
I rebased after https://github.com/gardener/gardener/pull/4379.

**Review by vpnachev — APPROVED** [review]
/lgtm
